### PR TITLE
Remove extensions to protobuf config checker messages

### PR DESCRIPTION
**Pull request by chbatey**
AFAICT these are never serialized/deserialized. Removing as they use a
deprecated feature of protobuf (required fields in extensions)

@renatocaval do you remember why we had these messages? Seems we only use the ConfigCheck message and check the type field

### CODE REVIEW

**Comment by chbatey** [comment]
Mima, will do if no one objects to removing these

**Review by jrudolph — APPROVED** [review]
LGTM afaiu

**Comment by renatocaval** [comment]
@chbatey, this is being used, here for instance:

https://github.com/akka/akka/pull/24120/files#diff-5d5f4d593392310ef98e95c33df9721aR68-R72

**Comment by renatocaval** [comment]
now, if we need to remove it because of deprecation we must find another solution and preferable something backward compatible.

**Review by raboof — COMMENTED** [review]
Given it looks like they're actually used (also over remoting) perhaps we should close this one and ticketify moving away from using `required` eventually?

**Comment by chbatey** [comment]
> https://github.com/akka/akka/pull/24120/files#diff-5d5f4d593392310ef98e95c33df9721aR68-R72

That is not the types from the generated proto, it is these plain types: https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala#L111

The only thing for the type of check that is used AFAICS is the type enum.

**Comment by chbatey** [comment]
I've added mima excludes. Unless I am cracking up these are not used. Would you mind checking again @renatocaval ?

**Review by octonato — APPROVED** [review]

**Comment by renatocaval** [comment]
@chbatey, you are right. I thought that we were using the generated typed, but we only use the enum type.

File: akka-cluster/src/main/protobuf/ClusterMessages.proto [hunk 1]
```diff
@@ -55,6 +55,7 @@ message InitJoinAck {
 }
 
 message ConfigCheck {
+  // Don't use extensions, removed in proto3
   extensions 100 to 102;
```
Comment (renatocaval) [inline akka-cluster/src/main/protobuf/ClusterMessages.proto]: Considering that this wasn't being used, do we still need it? Or can it be removed on future releases?

Comment (chbatey) [reply akka-cluster/src/main/protobuf/ClusterMessages.proto]: Yes I think we can as 100-102 have never been serialized but i'll do a quick test

Comment (jrudolph) [reply akka-cluster/src/main/protobuf/ClusterMessages.proto]: :+1: that would be great. I wondered about these but never used them in protobuf so kept quiet...

Comment (chbatey) [reply akka-cluster/src/main/protobuf/ClusterMessages.proto]: I've tested this and is fine to remove. The generated files don't include anything for extensions, only when creating an extension it validates that the field numbers are marked as extensions in the parent type

Comment (chbatey) [reply akka-cluster/src/main/protobuf/ClusterMessages.proto]: this isn't quite true, i was looking at protobuf3 generated code. Still works over the wire tho

**Comment by chbatey** [comment]
Merging this taking Renato's as the second review.

**Comment by patriknw** [comment]
👍

**Pull request merged**